sway-error/src/handler.rs

```diff
@@ -80,11 +80,40 @@ impl Handler {
         }
     }
 
+    pub fn append_ref(&self, other: &Handler) {
+        let other = other.inner.borrow();
+
+        for warn in other.warnings.iter() {
+            self.emit_warn(warn.clone());
+        }
+        for err in other.errors.iter() {
+            self.emit_err(err.clone());
+        }
+    }
+
```

Comment:
I don't think this is needed when you can just `h1.append(h2.clone())` and get the same result.

Reply:
They actually have a slight difference.

```
h1.append(h2.clone())
```

is cloning two `Vec`, two extra allocations. While `append_ref` will allocate only when the variants allocate (in some cases).

But I can simplify for now, given this is only called on error paths.